### PR TITLE
Zero assets bucket viacollateral

### PR DESCRIPTION
**Pull request by mattcushman**
An example bucket with 1 wei collateral and two LPs.  One redeems, leaving an empty bucket with positive LPB

### CODE REVIEW

**Comment by EdNoepel** [comment]
As a contributor, I cannot review this.

File: src/base/interfaces/pool/IPoolEvents.sol [hunk 1]
```diff
@@ -46,6 +46,16 @@ interface IPoolEvents {
         uint256 index
     );
 
+    /**
+     *  @notice Emitted when LPs are forfeited as a result of the bucket losing all assets.
+     *  @param  index       The index of the bucket.
+     *  @param  lpForfeited Amount of LP no forfeited by lenders.
```
Comment (MikeHathaway) [inline src/base/interfaces/pool/IPoolEvents.sol]: nit: remove "no" in param

File: src/libraries/external/Auctions.sol [hunk 2]
```diff
@@ -133,6 +135,16 @@ library Auctions {
         uint256 index
     );
 
+    /**
+     *  @notice Emitted when LPs are forfeited as a result of the bucket losing all assets.
+     *  @param  index       The index of the bucket.
+     *  @param  lpForfeited Amount of LP no forfeited by lenders.
```
Comment (MikeHathaway) [inline src/libraries/external/Auctions.sol]: same here; could also remove the console import

File: src/libraries/external/LenderActions.sol [hunk 3]
```diff
@@ -297,6 +307,8 @@ library LenderActions {
         fromBucket.lps -= fromBucketRedeemedLPs_;
         toBucket.lps   += toBucketLPs_;
 
+        // TODO: call _bankruptBucketIfEmpty?
```
Comment (MikeHathaway) [inline src/libraries/external/LenderActions.sol]: TODO: can be removed

File: tests/forge/ERC20Pool/ERC20PoolLiquidationsTake.t.sol [hunk 4]
```diff
@@ -2097,4 +2099,402 @@ contract ERC20PoolLiquidationsTakeTest is ERC20HelperContract {
             }
         );
     }
+
+    function testBucketZeroAssets() external  {
+        /*
+        _addInitialLiquidity(
```
Comment (MikeHathaway) [inline tests/forge/ERC20Pool/ERC20PoolLiquidationsTake.t.sol]: Can remove console import and unused `_addInitialLiquidity` function.

**Review by MikeHathaway — APPROVED** [review]
Had some comments about styling cleanups and dead comments to remove, but logic LGTM. Also has some merge conflicts with develop

File: src/base/interfaces/pool/IPoolEvents.sol (hunk 1 above)
Comment (EdNoepel) [reply src/base/interfaces/pool/IPoolEvents.sol]: done in 2609ba1f4fd84a2348093767cd9731d31c7d340e

File: src/libraries/external/Auctions.sol (hunk 2 above)
Comment (EdNoepel) [reply src/libraries/external/Auctions.sol]: done in 2609ba1f4fd84a2348093767cd9731d31c7d340e

File: src/libraries/external/LenderActions.sol (hunk 3 above)
Comment (EdNoepel) [reply src/libraries/external/LenderActions.sol]: done in 2609ba1f4fd84a2348093767cd9731d31c7d340e

File: tests/forge/ERC20Pool/ERC20PoolLiquidationsTake.t.sol (hunk 4 above)
Comment (EdNoepel) [reply tests/forge/ERC20Pool/ERC20PoolLiquidationsTake.t.sol]: done in 2609ba1f4fd84a2348093767cd9731d31c7d340e

**Comment by EdNoepel** [comment]
Not sure how to resolve these merge conflicts:
- It seems `testTakeAuctionPriceLtNeutralPrice`was replaced with `testBucketZeroAssets`; I don't know if that was intentional or both tests should exist independently.
- I'm unfamiliar with changes to `t0DebtPenalty_` and `poolDebt_` around line 1025.

Need @mattcushman to opine on this.

**Comment by grandizzy** [comment]
closing in favor of https://github.com/ajna-finance/contracts/pull/516

**Pull request closed without merging**